Utils/Dataflow/069_upload2es/load_data.sh

```diff
@@ -44,36 +67,65 @@ ES_PORT='9200'
 
 CURL_N_MAX=10
 SLEEP=5
-DELIMITER=`echo -e -n "\x00"`
-EOProcess=`echo -e -n "\x06"`
+EOBatch='\x17'
+BATCHMODE="d"
+
+case $BATCHMODE in
+  e|enabled)
+    [ -z "$EOB" ] && EOBatch="\n"
+    ;;
+  d|disabled)
+    ( [ -z "$EOB" ] || [ $EOB == "\x17" ] || [ $EOBatch == "\x17" ] ) && EOBatch="\n"
+    ( [ -n "$EOB" ] && [ ! $EOB == "\x17" ] ) && EOBatch="$EOB"
+    ;;
+  *)
+    log ERROR "Unexpected batch-mode parameter." && { [ -n "$BATCHMODE" ] && usage && return 2; }
+    break
+    ;;
+esac
```

Comment:
Not sure I\`m getting this `case` statement.
What makes me puzzled:
* `BATCHMODE` is just assigned with `"d"` (rewriting user value); why use `case $BATCHMODE`, when we already know the value for sure?
* `EOBatch` is just assigned with default value `'\x17'`; why compare it with `"\x17"` again?
* after this `case`, `EOBatch` will stay assigned with `'\x17'` only when batchmode is `enabled` and `EOB` is not empty;
* after this `case`, `EOBatch` will be assigned with value of `EOB` only when batchmode is `disabled` and `EOB` is not empty and not `"\x17"`.

What did you mean to do?